Test failure: FoundryClient connection lifecycle not working

## Issue Description

Integration test `should handle connection lifecycle` is failing with connection state issues.

## Error Details

```
AssertionError: expected false to be true // Object.is equality
- Expected: true
+ Received: false

❯ src/__tests__/integration.test.ts:91:36
  89| 
  90|       await expect(client.connect()).resolves.not.toThrow();
  91|       expect(client.isConnected()).toBe(true);
     |                                    ^
```

## Root Cause

The FoundryClient.connect() method is not properly setting the connection state to true after successful connection.

## Acceptance Criteria

- [ ] FoundryClient.connect() should set isConnected() to return true
- [ ] Integration test should pass consistently
- [ ] Connection state should be properly managed throughout lifecycle

--- a/src/__tests__/integration.test.ts
+++ b/src/__tests__/integration.test.ts
@@ -70,6 +70,7 @@ describe('Integration Tests', () => {
     it('should handle connection lifecycle', async () => {
       client = new FoundryClient({
         baseUrl: 'http://localhost:30000',
+        apiKey: 'test-key',
       });
 
       // Mock successful connection

--- a/src/foundry/client.ts
+++ b/src/foundry/client.ts
@@ -683,35 +683,42 @@ export class FoundryClient {
 
     const wsUrl = this.config.baseUrl.replace(/^http/, 'ws') + '/socket.io/';
 
-    try {
-      this.ws = new WebSocket(wsUrl);
+    return new Promise((resolve, reject) => {
+      try {
+        this.ws = new WebSocket(wsUrl);
 
-      this.ws.on('open', () => {
-        logger.info('WebSocket connected to FoundryVTT');
-      });
+        this.ws.on('open', () => {
+          this._isConnected = true;
+          logger.info('WebSocket connected to FoundryVTT');
+          resolve();
+        });
 
-      this.ws.on('message', (data) => {
-        try {
-          const message = JSON.parse(data.toString());
-          this.handleWebSocketMessage(message);
-        } catch (error) {
-          logger.warn('Failed to parse WebSocket message:', error);
-        }
-      });
+        this.ws.on('message', (data) => {
+          try {
+            const message = JSON.parse(data.toString());
+            this.handleWebSocketMessage(message);
+          } catch (error) {
+            logger.warn('Failed to parse WebSocket message:', error);
+          }
+        });
 
-      this.ws.on('error', (error) => {
-        logger.error('WebSocket error:', error);
-      });
+        this.ws.on('error', (error) => {
+          logger.error('WebSocket error:', error);
+          this.ws = null;
+          this._isConnected = false;
+          reject(error);
+        });
 
-      this.ws.on('close', () => {
-        logger.info('WebSocket disconnected');
-        this.ws = null;
-        this._isConnected = false;
-      });
-    } catch (error) {
-      logger.error('WebSocket connection failed:', error);
-      throw error;
-    }
+        this.ws.on('close', () => {
+          logger.info('WebSocket disconnected');
+          this.ws = null;
+          this._isConnected = false;
+        });
+      } catch (error) {
+        logger.error('WebSocket connection failed:', error);
+        reject(error);
+      }
+    });
   }
 
   /**

